scheduler/src/cook/mesos/api.clj

```diff
@@ -1371,16 +1367,26 @@
                        authorized? (fn [guuid] (is-authorized-fn user
                                                                  request-method
                                                                  {:owner (group-user guuid) :item :job}))
-                       unauthorized-guuids (mapv :uuid (remove authorized? guuids))]
-                   (if (empty? unauthorized-guuids)
-                     true
+                       {authorized-guuids true unauthorized-guuids false} (group-by authorized? guuids)]
+                   (if (or (empty? unauthorized-guuids) (::allow-partial-results? ctx))
+                     [true {::guuids (with-meta
+                                       authorized-guuids
+                                       ; Liberator will concatenate vectors by default when composing context values,
+                                       ; but we want to replace our vector with only the filtered entries.
+                                       {:replace true})}]
                      [false {::error (str "You are not authorized to "
                                           (case request-method
                                             :get "view"
                                             :delete "kill")
                                           " the following groups "
                                           (str/join \space unauthorized-guuids))}])))
-     :exists? (partial retrieve-groups conn)
+     :exists? (fn [ctx]
+                (or (::allow-partial-results? ctx)
+                    (empty? (::bad-uuids ctx))
+                    [false {::error (str "The following UUIDs didn't correspond to a group: "
+                                         (str/join
+                                           \space
+                                           (::bad-uuids ctx)))}]))
```

Comment:
I'm not following the logic here. Let's say `(::bad-uuids ctx)` is empty - wouldn't this cause `exists?` to return the error structure?

Reply:
The `or` here is equivalent to this:
```clojure
(cond  ; :exists?
  (::allow-partial-results? ctx) true
  (empty? (::bad-uuids ctx)) true
  :else [false {:error ...}])